#include "stdafx.h"\n#include < NUnit.Framework >\nusing namespace System;\n// or whatever your using directive for NUnit is\n\n[TestFixture]\npublic ref class ExceptionTests {\npublic:\n    [Test]\n    void Test_Something_That_Throws_Exception() {\n        try {\n            throw gcnew ArgumentOutOfRangeException("Some more detail");\n        }\n        catch (ArgumentOutOfRangeException^ ex) {\n            Assert::IsInstanceOf<ArgumentOutOfRangeException^>(ex);\n            // Optionally, you can also check the exception message\n            Assert::AreEqual("Some more detail", ex->ParamName);\n            return; // If the expected exception is caught, return from test method\n        }\n        catch (...) {\n            Assert::Fail("Unexpected exception type was thrown.");\n        }\n\n        Assert::Fail("Expected ArgumentOutOfRangeException to be thrown.");\n    }\n};